framework/core/command_buffer.cpp

```diff
@@ -712,11 +712,22 @@ void CommandBuffer::flush_descriptor_state(VkPipelineBindPoint pipeline_bind_poi
 				}
 			}
 
-			// Request a descriptor set from the render frame, and write the buffer infos and image infos of all the specified bindings
-			auto &descriptor_set = command_pool.get_render_frame()->request_descriptor_set(descriptor_set_layout, buffer_infos, image_infos, command_pool.get_thread_index());
-			descriptor_set.update(bindings_to_update);
-
-			VkDescriptorSet descriptor_set_handle = descriptor_set.get_handle();
+			VkDescriptorSet descriptor_set_handle;
+			if (command_pool.get_render_frame()->get_descriptor_management_strategy() == DescriptorManagementStrategy::StoreInCache)
+			{
+				// Request a descriptor set from the render frame, and write the buffer infos and image infos of all the specified bindings
+				auto &descriptor_set = command_pool.get_render_frame()->request_descriptor_set(descriptor_set_layout, buffer_infos, image_infos, command_pool.get_thread_index());
+				descriptor_set.update(bindings_to_update);
+				descriptor_set_handle = descriptor_set.get_handle();
+			}
+			else
+			{
+				// Request a descriptor pool, allocate a descriptor set, write buffer and image data to it
+				auto &        descriptor_pool = command_pool.get_render_frame()->request_descriptor_pool(descriptor_set_layout, command_pool.get_thread_index());
+				DescriptorSet descriptor_set{command_pool.get_render_frame()->get_device(), descriptor_set_layout, descriptor_pool, buffer_infos, image_infos};
+				descriptor_set.apply_writes();
+				descriptor_set_handle = descriptor_set.get_handle();
+			}
```

Comment:
All operations here are done via `command_pool.get_render_frame()->`.
Looks like this functionality should be moved into `vkb::RenderFrame`, such that the `vkb::CommandBuffer` doesn't even need to know about the `DescriptorManagementStrategy`?

Reply:
Good point, moved it

Reply:
Now that this functionality has been moved, it reveals a potential other issue: those `bindings_to_update` are used in the `StoreInCache`-case only, but determined in `CommandBuffer::flush_descriptor_state` independently of that state.
Can that vector be determined from the data passed into `RenderFrame::request_descriptor_set`, if required only?

Reply:
Moved collecting `bindings_to_update` to `RenderFrame::request_descriptor_set`. Unfortunately, I had to add a new function `collect_bindings_to_update` because  `bindings_to_update` have to be aggregated from `image_infos` and `buffer_infos`.
The complexity of `collect_bindings_to_update` doesn't affect performance, in `constant_data` sample there are only 3 `binding_index`s to check overall, and this number would not likely ever change significantly.

Reply:
Looks good now.
Just one question: Previously, each `binding_index` (given some conditions) from the `resource_set` was added to `bindings_to_update`. Now each `binding_index` (given the same conditions) from the `buffer_infos` and `image_infos` maps are used. Obviously, in this sample that is the same, but does that hold in general? Could that somehow be documented in code (by some assertions or such)?

Reply:
Yes, it can. It's easy to check if a `binding_index` from the `resource_set` ended up as a key in either `buffer_infos` or `image_infos`. If it did, it would be evaluated in `collect_bindings_to_update`.

As before, I tested the code by running the `constant_data` sample (the only sample that sets `update_after_bind` to `true`) and the assertion doesn't fail.

In addition, I removed `std::move`, since both `VkDescriptorImageInfo` and `VkDescriptorBufferInfo` are trivial structs that can't benefit from move semantics.

Reply:
Just one very minor thing that caught my attention now: in
`else if (image_view != nullptr || sampler != VK_NULL_HANDLE)`
you should compare sampler with nullptr as well, as it's just a pointer and not a handle.

Reply:
fixed